azure/interfaces.go

```diff
@@ -127,3 +128,13 @@ type ResourceSpecGetterWithHeaders interface {
 	// CustomHeaders returns the headers that should be added to Azure API calls.
 	CustomHeaders() map[string]string
 }
+
+// ASOResourceSpecGetter is an interface for getting all the required information to create/update/delete an Azure resource.
+type ASOResourceSpecGetter interface {
+	// ResourceRef returns a concrete, named (and namespaced if applicable) ASO
+	// resource type to facilitate a strongly-typed GET.
+	ResourceRef() genruntime.MetaObject
+	// Parameters returns a modified object if it points to a non-nil resource.
+	// Otherwise it returns a new value.
```

Comment:
can this comment also define the contract for "no updates needed"? i.e. return nil or return the same unmodified object

Reply:
Yep I'll add that.